sources/packages/api/src/route-controllers/student-appeal/student-appeal.students.controller.ts

```diff
@@ -0,0 +1,94 @@
+import {
+  Controller,
+  Param,
+  Post,
+  Body,
+  NotFoundException,
+  UnprocessableEntityException,
+} from "@nestjs/common";
+import {
+  StudentAppealService,
+  ApplicationService,
+  FormService,
+} from "../../services";
+import { StudentAppealDTO } from "./models/student-appeal.dto";
+import { AuthorizedParties } from "../../auth/authorized-parties.enum";
+import { AllowAuthorizedParty, UserToken } from "../../auth/decorators";
+import { IUserToken } from "../../auth/userToken.interface";
+import {
+  ApiTags,
+  ApiCreatedResponse,
+  ApiNotFoundResponse,
+  ApiUnprocessableEntityResponse,
+} from "@nestjs/swagger";
+import BaseController from "../BaseController";
+import { ClientTypeBaseRoute, ApiProcessError } from "../../types";
+import { INVALID_APPLICATION_NUMBER } from "../../constants";
+
+@AllowAuthorizedParty(AuthorizedParties.student)
+@Controller("appeal")
+@ApiTags(`${ClientTypeBaseRoute.Student}-appeal`)
+export class StudentAppealStudentsController extends BaseController {
+  constructor(
+    private readonly studentAppealService: StudentAppealService,
+    private readonly applicationService: ApplicationService,
+    private readonly formService: FormService,
+  ) {
+    super();
+  }
+
+  /**
+   * Submit student appeal.
+   * @param applicationId application for which the appeal is submitted.
+   * @param payload student appeal with appeal requests.
+   * @param userToken
+   */
+  @ApiCreatedResponse({ description: "Student appeal created successfully" })
+  @ApiNotFoundResponse({
+    description:
+      "Application either not found or not eligible to request for change.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description: "There is an existing appeal for this student.",
+  })
+  @Post("application/:applicationId")
+  async submitStudentAppeal(
+    @Param("applicationId") applicationId: number,
+    @Body() payload: StudentAppealDTO,
+    @UserToken() userToken: IUserToken,
+  ): Promise<void> {
+    const application = this.applicationService.getApplicationToRequestChange(
+      undefined,
+      userToken.userId,
+      applicationId,
+    );
+    if (!application) {
+      throw new NotFoundException(
+        new ApiProcessError(
+          "Given application either does not exist or is not complete to request change.",
+          INVALID_APPLICATION_NUMBER,
+        ),
+      );
+    }
+    const existingStudentAppeal =
+      await this.studentAppealService.hasExistingAppeal(userToken.userId);
+    if (existingStudentAppeal) {
+      throw new UnprocessableEntityException(
+        "There is already a pending appeal for this student.",
+      );
+    }
+    const submissionResults = payload.studentAppealRequests.map(
+      (studentAppeal) =>
+        this.formService.dryRunSubmission(
+          studentAppeal.formName,
+          studentAppeal.formData,
+        ),
+    );
+    await Promise.all(submissionResults);
```

Comment:
Is Promise.all needed here since you are not trying to resolve an array of results ?

Reply:
Actually we are trying resolve the results but we don't parse the response. While submitting student appeal request, each appeal is a form and we are doing dryRunSubmission for array of forms and waiting for the results of all the forms. Until dryrun is successfully completed for all the forms, we do not move further. 
So await of promise.all is required here.

Reply:
I like the way that it is executed in parallel, great work, but we do need to inspect the results, like the below, right?
![image](https://user-images.githubusercontent.com/61259237/159755407-1c7f55ad-4716-4b8b-9779-f1d657b92a09.png)

Reply:
@andrewsignori-aot Yes. that is missed. will add that part.